src/vector/layer.rs

```diff
@@ -467,6 +470,43 @@ pub trait LayerAccess: Sized {
             gdal_sys::OGR_L_SetAttributeFilter(self.c_layer(), null_mut());
         }
     }
+
+    /// Read batches of columnar [Arrow](https://arrow.apache.org/) data from OGR
+    ///
+    /// Refer to the example provided in `read_ogr_arrow.rs`.
+    ///
+    /// This API is new as of GDAL 3.6.
+    ///
+    /// # Safety
+    /// This uses the Arrow C Data Interface to operate on raw pointers provisioned from Rust.
+    /// These pointers must be valid and provisioned according to the ArrowArrayStream spec
+    #[cfg(any(major_ge_4, all(major_is_3, minor_ge_6)))]
+    unsafe fn read_arrow_stream(
+        &mut self,
+        out_stream: *mut gdal_sys::ArrowArrayStream,
+    ) -> Result<()> {
+        self.reset_feature_reading();
+
+        let options = std::ptr::null_mut();
+
+        unsafe {
+            // Note to self: example of how to operate on options
```

Comment:
There's a nicer way to do this: https://github.com/georust/gdal/commit/a68aba7359dab2cceddddc3a4f3022ad902bfa5d#diff-735b4e8f56e7d6ad6abe40c99808df64a2b2e05b2b13f976303cefe2dca42affR174.

You might want to remove or update the commented-out code.

Reply:
Yeah, that's a good idea.

Reply:
I changed it to `Option<&CslStringList>`

Reply:
Let's drop the `Option`, `Geometry::make_valid` ended up not using one. An empty `CslStringList` gets passed as a null pointer, so there shouldn't be any downsides.

And can you squash the commits if it's not too much of a hassle?

Reply:
I took out the `Option` and squashed